lib/net.js

```diff
@@ -1018,21 +1020,25 @@ function lookupAndConnect(self, options) {
   var localPort = options.localPort;
 
   if (localAddress && !cares.isIP(localAddress)) {
-    throw new TypeError('"localAddress" option must be a valid IP: ' +
-                        localAddress);
+    throw new errors.Error('ERR_INVALID_IP_ADDRESS', localAddress);
```

Comment:
Why was this changed to a `Error`? It was a `TypeError` before?

Reply:
I actually think not using `TypeError` is appropriate here. I've never felt that `TypeError` was a good fit.

Reply:
my bad. I corrected this, now the error types in lib/net.js have the correct value.

Reply:
@jasnell i have already committed (some minutes ago) a change that preserve the error types as they were in `lib/net.js`. So this is now a `TypeError` as before. Should I re-changed it?